Objects/longobject.c

```diff
@@ -4082,8 +4094,11 @@ l_divmod(PyLongObject *v, PyLongObject *w,
         return 0;
     }
 #if WITH_PYLONG_MODULE
-    if (Py_ABS(Py_SIZE(w)) > 1000) {
-        /* Switch to _pylong.int_divmod() */
+    Py_ssize_t size_w = Py_ABS(Py_SIZE(w));
+    Py_ssize_t size_v = Py_ABS(Py_SIZE(v));
+    if (size_w > 1000 && (size_w - size_v) > 1) {
```

Comment:
```suggestion
    if (size_w > 1000 && (size_v - size_w) > 1) {
```
Swapped the order of the operands. `v` is the numerator, right? We want to go to Python if the numerator is "much" larger than the denominator, not the opposite.